### ISSUE TITLE
downloading workspace data on safari

### ISSUE THREAD

**Issue by nconrad** [PATRIC3/patric3_website#1159]
None of the pre-auth download buttons work for me in safari.  This is because when you click download button, the `Workspace.get_download_url` is called, and then the browser opens that url in a new tab to start the download.  Should I switch to fetching the urls and putting them on the page as download links?  This will also allow right-click > "save as" to work.

In firefox, I also get the annoying:

![image](https://cloud.githubusercontent.com/assets/1194246/19581508/18ff1736-96f3-11e6-9bf1-8f03c8dff048.png)

**Comment by nconrad** [comment]
ah, so one concern about making this switch is that I think it can take up to a couple seconds for the download_service to respond, at least when fetching several urls.

**Comment by olsonanl** [comment]
Say we generate a download URL that is directly linkable; can we ensure that auth information is included (via cookie perhaps)? If so we can make the requirement to call get_download_url go away.

**Comment by nconrad** [comment]
@olsonanl and I discussed this a little more on Friday and think what he is suggesting is a good idea.  I'll have to talk to @dmachi more about the cookie changes this week.

**Comment by dmachi** [comment]
For the other downloads to the API, I add the token into the url like &http_authorization=TOKEN.  The disadvantage of this is that some proxy server might log the full url with the token in it and since our tokens live for a year, that wouldn’t be great.  Other than this logging, its not really any less secure than putting the token in a cookie.

**Comment by dmachi** [comment]
Yes it was done that way because it was a separate call to get the downloadable url and that took time (and also the download urls only live for an hour or so).  All that said, you can probably change things to simply not open in a new browser tab/window now.  It was mainly like that before because there were a lot of errors in the download and I couldn’t control the error display.  I think most of those issues have been addressed now, so opening in the same window/tab is fine.

**Comment by mshukla1** [comment]
@dmachi I guess this is not critical issue, agree?

**Comment by mshukla1** [comment]
I tried several downloads in safari from the feature table and they worked fine. So, I am not sure which download problem this refers to. 

I am marking this non critical. Let me know if anyone disagrees.

**Comment by nconrad** [comment]
It's an issue with downloading workspace data via the workspace browser or workspace object pages.  The SOLR data downloads should work fine.

**Comment by zreyn** [comment]
Looking into this, the recent Job Result viewer changes don't materially change the way the links are generated.  

Here's where they are generated:
1) https://github.com/PATRIC3/p3_web/blob/cea6032ef32fdb34305f4360e2fc20dc8a749217/public/js/p3/widget/WorkspaceBrowser.js#L270-L277

2) https://github.com/PATRIC3/p3_web/blob/cea6032ef32fdb34305f4360e2fc20dc8a749217/public/js/p3/WorkspaceManager.js#L348-L353

I verified that we are pulling the proper download URLs.  The difference for me was that the window.open call is blocked on Safari by default, so clicking on the link appears to do nothing.  Turning off the option Safari->Preferences->Security->Block Pop-up Windows lets Safari work properly.

Swapping window.location.assign(url) vs window.open(url) seems to do the trick.  Fix included in the following PR:
https://github.com/PATRIC3/p3_web/pull/506

**Comment by olsonanl** [comment]
does this mean the downloads no longer trigger the extra empty window and popup warnings? If so that is excellent news. that's always been an really annoying feature.

**Comment by olsonanl** [comment]
BTW if we can get the cookie issue determined we can indeed make the extra call to get download url go away except in the cases where one wants to export a workable download link to something that won't/can't provide the cookie.

**Comment by zreyn** [comment]
The immediate issue here has been fixed, so I will close this issue.  If we want to go after the cookie solution, let's take that to another issue.
